Funcionalidad de carrito

DanyElAlgo opened the pull request:
Refactorizando el método de obtención de carrito para no depender de un mockup, sino obtener de forma real los productos al "añadir un producto al carrito de pedido". Aún falta realizar una forma de que no se reinicie el carrito al recargar la página y de mostrar el ícono de carrito en el navbar en vez de la página del restaurante.

DanyElAlgo committed:
Funcionalidad de carrito

Refactorizando el método de obtención de carrito para no depender de un mock, sino obtener de forma real los productos al "añadir un producto al carrito de pedido"

DanyElAlgo committed:
correcciones al texto de carrito

esto no es el estilo final, es únicamente para debug y se eliminará en futuros commits

DanyElAlgo commented:
![Imagen de WhatsApp 2025-04-22 a las 15 00 06_7110dc62](https://github.com/user-attachments/assets/1bf0a929-66a8-4158-a01d-495c3603bcba)
Mockup y detalles a considerar de la funcionalidad del carrito

## TODO:
- Llevar el ícono de carrito al navbar
- Alterar el ícono dependiendo si tiene o no productos
- Reconfigurar el "Create", para verificar si existen duplicados

Opcional: Hacer que el carrito se guarde en una cookie para no reiniciarse al refrescar la página

DanyElAlgo committed:
Funcionalidad de carrito mejorada

Ya detecta productos duplicados, ahora solo falta llevarlo al navbar en vez de seguir en la página del restaurante

DanyElAlgo committed:
funcionalidad de carrito completada

ya agarra bien todos los productos, es individual por restaurante, lo que significa que se reinicia al cambiar de restaurante, también desaparece cuando no hay items elegidos o cuando ya se realizó el pedido en el checkout

RaidenMT commented on OrderNow-React/src/components/ProductDetailsModal.jsx:
```diff
@@ -1,17 +1,52 @@
-    // TODO: Aqui se tiene que manejar la logica para mandar el producto al carrito
-    console.log("Producto seleccionado:", product, "Cantidad:", quantity);
+    {
+      if (products != null) {
```
Refactorizar este código

RaidenMT commented on OrderNow-React/src/components/layout/NavBar.jsx:
```diff
@@ -1,73 +1,173 @@
+    navigate(`/restaurante/${restaurantId}/cart`);
+  };
+
+  const showCart = getQuantity > 0; // Example condition to show the cart link
```
Quitar comentarios

RaidenMT commented on OrderNow-React/src/components/layout/NavBar.jsx:
```diff
@@ -1,73 +1,173 @@
+                  stroke="currentColor"
+                  class="h-8 w-8 text-white"
+                >
+                  <path
```
Me imagino este es el ícono del carrito. Cuando el tema es light no se ve el ícono. Arreglar

![image](https://github.com/user-attachments/assets/d880cfca-99d6-4389-9fdf-8028487e0f5d)

RaidenMT commented on OrderNow-React/src/main.jsx:
```diff
@@ -4,7 +4,6 @@ import { SuperTokensConfig } from './Supertokens.jsx'
 import App from './App.jsx'
 import './index.css'
 
-SuperTokensConfig();
```
¿Por qué se hace esto? Justificar

DanyElAlgo replied:
En este caso debí comentar la línea en vez de eliminarla, lo hice porque falta un archivo de configuración para los supertokens y eso es parte de Henrry, no va para mi caso, y la presencia de la línea impide que el programa inicie

RaidenMT commented on OrderNow-React/src/context/AddToCart.js:
Qué pasó ocn este archivo

DanyElAlgo replied:
El archivo era una especie de placeholder de cómo implementar un botón que introduzca artículos al carrito, sin embargo, tras revisar la funcionalidad del menú del restaurante, fue mejor implementar dicha funcionalidad directamente allí, dejando este placeholder ya sin uso.

DanyElAlgo committed:
Correcciones realizadas

Se corrigieron todos los detalles remarcados en la review del anterior commit, ya debe estar completamente listo

RaidenMakio commented on OrderNow-React/src/components/ProductDetailsModal.jsx:
```diff
@@ -1,17 +1,43 @@
+      if(products != null){
+        existingProduct = products.find((p) => p.id === product.id);
+      }
+      if (existingProduct) {
```
Refactorizar para no usar if-else

RaidenMakio commented on OrderNow-React/src/components/ProductDetailsModal.jsx:
```diff
@@ -1,17 +1,43 @@
-    // TODO: Aqui se tiene que manejar la logica para mandar el producto al carrito
-    console.log("Producto seleccionado:", product, "Cantidad:", quantity);
+    {
+      var existingProduct = null;
```
Quitar var y refactorizar para no usar if

DanyElAlgo committed:
Última refactorización terminada

Adiós if-else, hola ternarios

DanyElAlgo commented:
Última refactorización terminada! https://github.com/ucb-scz-dims/SIS-213-ON/pull/38/commits/c780e0fa1fecde3a31cedd16a8456c654fa68c6b
